apis/Google.Cloud.Bigtable.Admin.V2/Google.Cloud.Bigtable.Admin.V2.Tests/IamVersionTest.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Iam.V1;
+using Grpc.Core;
+using System;
+using Xunit;
+
+namespace Google.Cloud.Bigtable.Admin.V2.Tests
+{
+    public class IamVersionTest
+    {
+        [Theory]
+        [InlineData(null, 1, 1)]
+        [InlineData(null, null, null)]
+        [InlineData(3, 1, 1)]
+        [InlineData(3, null, 3)]
+        [InlineData(1, 3, 3)]
+        [InlineData(1, null, 1)]
```

Comment:
settingsVersion - set by the client library at the point of release.
callerRequestVersion - The version a user provides when performing a GetIamPolicy Request and meant to override the settingsVersion if provided.
expectedRpcVersion - Is the version the client is sending to the service.

Could you please confirm these definitions?

Reply:
Not quite. `settingsVersion` is taken from the settings used to create the instance of the client. The *default* is set at the point of release, but the user can specify settings themselves - just as we do in this test. This allows them to create a client which will *always* use version 3, for example (modulo per-request settings)... whereas the `callerRequestVersion` is specified just for a *single* request.